tests/unit/dogstatsd/test_statsd.py

```diff
@@ -767,6 +767,61 @@ def test_entity_tag_and_tags_from_environment_and_constant(self):
         tags = "country:canada,red,country:china,age:45,blue,dd.internal.entity_id:04652bb7-19b7-11e9-9cc6-42010a9c016d"
         assert_equal_telemetry('gt:123.4|g|#'+tags, statsd.socket.recv(), telemetry=telemetry_metrics(tags=tags))
 
+    def test_dogstatsd_initialization_with_dd_env_service_version(self):
+        """
+        Dogstatsd will automatically use DD_ENV, DD_SERVICE, and DD_VERSION
+        to set the {env, service, version} tags for all metrics emitted.
+
+        This test covers several possible cases of tag configuration.
+        Tags configured from any source are added in an append-only fashion.
+        """
+        cases = [
+            ('prod', 'dog', 'abc123', '', [], ['env:prod', 'service:dog', 'version:abc123']),
+            ('prod', 'dog', 'abc123', 'env:prod,type:app', [], ['env:prod', 'type:app', 'env:prod', 'service:dog', 'version:abc123']),
+            ('prod', 'dog', 'abc123', 'env:prod2,type:app', [], ['env:prod2', 'type:app', 'env:prod', 'service:dog', 'version:abc123']),
+            ('prod', 'dog', 'abc123', '', ['env:prod', 'type:app'], ['env:prod', 'type:app', 'env:prod', 'service:dog', 'version:abc123']),
+            ('prod', 'dog', 'abc123', '', ['env:prod2', 'type:app'], ['env:prod2', 'type:app', 'env:prod', 'service:dog', 'version:abc123']),
+            ('prod', 'dog', 'abc123', 'env:prod3,custom_tag:cat', ['env:prod2', 'type:app'], ['env:prod2', 'type:app', 'env:prod3', 'custom_tag:cat', 'env:prod', 'service:dog', 'version:abc123']),
+        ]
+        for c in cases:
+            dd_env, dd_service, dd_version, datadog_tags, constant_tags, global_tags = c
+            with (
```

Comment:
I am sorry it can't be a "tuple" in this case. Can we update the `preserve_environment_variable` function to accept multiple values?